tools/slang-unit-test/unit-test-record-replay.cpp

```diff
@@ -0,0 +1,437 @@
+    SlangResult res = launchProcessAndReadStdout(context, optArgs, "slang-replay", process, exeRes);
+    disableLogInReplayer();
+
+    if (SLANG_FAILED(res))
+    {
+        return res;
+    }
+
+    List<String> hashLines;
+    for (auto line : LineParser(exeRes.standardOutput.getUnownedSlice()))
+    {
+        if (line.getLength() == 0)
+        {
+            continue;
+        }
+
+        if (line.indexOf(UnownedStringSlice("hash:")) == -1)
+        {
+            continue;
+        }
+
+        hashLines.add(line);
+    }
+
+    res = parseHashes(hashLines, outHashes);
+    if (SLANG_FAILED(res))
+    {
+        msgBuilder << "Failed to parse hash from stdout of 'slang-replay'\n";
+        getTestReporter()->message(TestMessageType::TestFailure, msgBuilder.toString().getBuffer());
+        return SLANG_FAIL;
+    }
+
+    return SLANG_OK;
+}
+
+static SlangResult resultCompare(List<entryHashInfo> const& expectHashes, List<entryHashInfo> const& resultHashes)
+{
+    if (expectHashes.getCount() != resultHashes.getCount())
+    {
+        return SLANG_FAIL;
+    }
+
+    StringBuilder msgBuilder;
+    for (Index i = 0; i < expectHashes.getCount(); i++)
+    {
+        if (expectHashes[i].targetIndex != resultHashes[i].targetIndex)
+        {
+            msgBuilder << "Failed to match 'targetIndex' at index " << i << "\n";
+            msgBuilder << "Expect: " << expectHashes[i].targetIndex << ", actual: " << resultHashes[i].targetIndex << "\n";
+            getTestReporter()->message(TestMessageType::TestFailure, msgBuilder.toString().getBuffer());
+            return SLANG_FAIL;
+        }
+        if (expectHashes[i].entryPointIndex != resultHashes[i].entryPointIndex)
+        {
+            msgBuilder << "Failed to match 'entryPointIndex' at index " << i << "\n";
+            msgBuilder << "Expect: " << expectHashes[i].entryPointIndex << ", actual: " << resultHashes[i].entryPointIndex << "\n";
+            getTestReporter()->message(TestMessageType::TestFailure, msgBuilder.toString().getBuffer());
+            return SLANG_FAIL;
+        }
+
+        if (expectHashes[i].hash != resultHashes[i].hash)
+        {
+            msgBuilder << "Failed to match 'hash' at index " << i << "\n";
+            msgBuilder << "Expect: " << expectHashes[i].hash << ", actual: " << resultHashes[i].hash << "\n";
+            getTestReporter()->message(TestMessageType::TestFailure, msgBuilder.toString().getBuffer());
+            return SLANG_FAIL;
+        }
+    }
+
+    return SLANG_OK;
+}
+
+static SlangResult cleanupRecordFiles()
+{
+    if (File::exists("slang-record") == false)
+    {
+        return SLANG_OK;
+    }
+
+    StringBuilder msgBuilder;
+    // Path::remove() doesn't support remove a non-empty directory, so we need to implement
+    // a simple function to remove the directory recursively.
+#ifdef _WIN32
+    // https://learn.microsoft.com/en-us/windows/win32/api/shellapi/nf-shellapi-shfileoperationa
+    SHFILEOPSTRUCTA file_op = {
```

Comment:
Might be good to put this logic inside `Path::remove`.

Reply:
Sure, let me create a separate small issue for this. (#4916 )